app/src/ui/branches/branches-container.tsx

```diff
@@ -275,10 +303,37 @@ export class BranchesContainer extends React.Component<
         dispatcher={this.props.dispatcher}
         repository={repository}
         isLoadingPullRequests={this.props.isLoadingPullRequests}
+        onMouseEnterPullRequest={this.onMouseEnterPullRequestListItem}
+        onMouseLeavePullRequest={this.onMouseLeavePullRequestListItem}
       />
     )
   }
 
+  private onMouseEnterPullRequestListItem = (
+    pullRequestBeingViewed: PullRequest
+  ) => {
+    this.setState({ pullRequestBeingViewed })
+  }
+
+  private onMouseLeavePullRequestListItem = async (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>
+  ) => {
+    // If we leave a list item, onto the pull request quick view, we don't want
+    // to close the quick view
+    const { relatedTarget } = event
+    const prQuickView = document.getElementById('pull-request-quick-view')
```

Comment:
Could we solve this by using a Ref instead of relying on a global id?

If we add a prop like this to the QuickView component

```
  readonly rootElementRef?: React.Ref<HTMLDivElement>
```

We can pass that to the `ref` attribute of the root div inside the quickview and use that. In "modern" react you'd use `React.forwardRef` (https://reactjs.org/docs/forwarding-refs.html)

Reply:
Neat! I will try that out.